persistence-api/src/main/java/io/stargate/db/query/builder/AbstractBound.java

```diff
@@ -0,0 +1,76 @@
+package io.stargate.db.query.builder;
+
+import com.datastax.oss.driver.shaded.guava.common.annotations.VisibleForTesting;
+import com.datastax.oss.driver.shaded.guava.common.base.Preconditions;
+import io.stargate.db.Parameters;
+import io.stargate.db.datastore.ResultSet;
+import io.stargate.db.query.AsyncQueryExecutor;
+import io.stargate.db.query.BoundQuery;
+import io.stargate.db.query.QueryType;
+import io.stargate.db.query.TypedValue;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.UnaryOperator;
+import org.apache.cassandra.stargate.db.ConsistencyLevel;
+
+// In theory, we should access this outside of this package. However, some test cast a BoundQuery
+// into this class to reach into the underlying BuiltQuery, and this out of convenience. This could
+// be made package private by changing those tests, but it's probably not a big deal.
+@VisibleForTesting
+public class AbstractBound<Q extends BuiltQuery<?>> implements BoundQuery {
+  private final Bounded<Q> bounded;
+  private final List<TypedValue> values;
+
+  AbstractBound(Q builtQuery, List<TypedValue> boundedValues, List<TypedValue> values) {
+    this.bounded = new Bounded<>(builtQuery, boundedValues);
+    this.values = values;
+  }
+
+  @Override
+  public QueryType type() {
+    return bounded.query().type();
+  }
+
+  @Override
+  public Bounded<Q> bounded() {
```

Comment:
You're right, I messed that one.

Reply:
Fwiw, I renamed it to `Source` (I removed the direct import, so it's used as `BoundQuery.Source`). I'm wary of calling it `Bound` because this is part of `BoundQuery` and I feel this gets confusing. Thats btw why my brain came up with "bounded": I was trying to convey a subtle difference between the what had been bound together, and the result of it. But I didn't realized I was changing the meaning to something else in doing so.

Anyway, since this truly represents "the query and values that were bound together to obtain the BoundQuery", I figured `BoundQuery.Source` is not too bad. If a better naming occurs to you, please do share.